src/categorical_algebra/FinCats.jl

```diff
@@ -331,6 +332,47 @@ FinFunctor(ob_map, hom_map, dom::Presentation, codom::Presentation) =
 Categories.show_type_constructor(io::IO, ::Type{<:FinFunctor}) =
   print(io, "FinFunctor")
 
+function is_initial(F::FinFunctor)::Bool
+  Gₛ, Gₜ = F.dom.graph, F.codom.graph
```

Comment:
An instance of the abstract type `FinFunctor` is not guaranteed to have `dom` and `codom` fields, e.g. an identity functor has no `codom` field, so you should use the accessor functions, which always work:

```julia
Gₛ, Gₜ = graph(dom(F)), graph(codom(F))
```